pdns/pdns_recursor.cc

```diff
@@ -1596,6 +1606,21 @@ static bool answerIsNOData(uint16_t requestedType, int rcode, const std::vector<
   return true;
 }
 
+static bool isAllowNotifyForZone(DNSName qname)
+{
+  if (t_allowNotifyFor->empty()) {
+    return false;
+  }
+
+  notifyset_t::const_iterator ret;
+  do {
+    ret = t_allowNotifyFor->find(qname);
+    if (ret != t_allowNotifyFor->end())
+      break;
```

Comment:
nit: you could `return true` here and `return false` after `while`

Reply:
Nice! Done.